docs/rfcs/XXX-pull-request-review.md

```diff
@@ -0,0 +1,145 @@
+# Feature title
+
+Pull Request Review
+
+## Status
+
+Proposed
+
+## Summary
+
+Give and receive code reviews on pull requests within Atom.
+
+## Motivation
+
+Workflows around pull request reviews involve many trips between your editor and your browser. If you check out a pull request locally to test it and want to leave comments, you need to map the issues that you've found in your working copy back to lines on the diff to comment appropriately. Similarly, when you're given a review, you have to mentally correlate review comments on the diff on GitHub with the corresponding lines in your local working copy, then map _back_ to diff lines to respond once you've established context. By revealing review comments as decorations directly within the editor, we can eliminate all of these round-trips and streamline the review process for all involved.
+
+Peer review is also a critical part of the path to acceptance for pull requests in many common workflows. By surfacing progress through code review, we provide context on the progress of each unit of work alongside existing indicators like commit status.
+
+## Explanation
+
+### Current pull request tile
+
+Reviews on the current pull request are rendered as a list on the current pull request tile.
+
+![review-list](https://user-images.githubusercontent.com/378023/44708426-1f582000-aae2-11e8-86bd-3074ae259e2d.png)
+
+* The review summary bubble is elided after the first sentence or N characters if necessary.
+* Clicking the review summary bubble opens an `IssueishPaneItem` in the workspace center, open to the reviews tab.
+* Clicking a line comment opens or activates an editor on the referenced file and scrolls to center the comment's line, translated according to local changes if appropriate.
+* Line comments within the review are rendered: _with a dot_ before the file has been opened and the corresponding decoration is visible; _with no icon_ after the file and decoration have been seen; and _with a checkmark_ after the comment has been marked "resolved" with the control on its decoration.
+
+If a new review has been started locally, it appears at the top of the "Reviews" section within this tile:
```

Comment:
A limitation with submitting a review from "Current pull request" is that you can only review a PR if you do a checkout first. Is that ok? Quality of reviews will probably go up, but quantity go down. Hmm.. 🤔. I'll see how an alternative feels under the "Changes" tab, since it's where you can see the full diff.

Reply:
Oh, that's true. It'd be better to offer the ability to submit PRs from the "Reviews" tab in the IssueishPaneItem, too (replicating the same UX). Basically anywhere we show reviews, we should also show your pending review and give you a way to finalize it.

Reply:
> That said, if we're going to support reviewing a PR for a branch that does not match your locally checked out branch, we'll have to consider how to handle inline review comments.

We _could_ implicitly check out the PR when you click `<>`... I'm leaning toward just disabling the button, though, too :smile: